### PR TITLE
Add HuaweiCloud Storing Support

### PR DESCRIPTION
**Pull request by vincywindy**
https://github.com/vincywindy/Volo.Abp.BlobStoring.HuaweiCloud

### CODE REVIEW

**Comment by maliming** [comment]
hi

Can you convert to file scoped namespace and format code?

**Comment by vincywindy** [comment]
> hi
> 
> Can you convert to file scoped namespace and format code?

I wrote it in Minio format

File: framework/src/Volo.Abp.BlobStoring.HuaweiCloud/HuaweiCloudBlobProvider.cs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+namespace Volo.Abp.BlobStoring.HuaweiCloud
+{
+    public class HuaweiCloudBlobProvider : BlobProviderBase, ITransientDependency
+    {
+        protected IHuaweiCloudBlobNameCalculator HuaweiCloudBlobNameCalculator { get; }
+        protected IBlobNormalizeNamingService BlobNormalizeNamingService { get; }
+
+        public HuaweiCloudBlobProvider(
+            IHuaweiCloudBlobNameCalculator HuaweiCloudBlobNameCalculator,
```
Comment (rqx110) [inline framework/src/Volo.Abp.BlobStoring.HuaweiCloud/HuaweiCloudBlobProvider.cs]:
```suggestion
            IHuaweiCloudBlobNameCalculator huaweiCloudBlobNameCalculator,
```

File: framework/src/Volo.Abp.BlobStoring.HuaweiCloud/HuaweiCloudBlobProvider.cs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+            IBlobNormalizeNamingService blobNormalizeNamingService)
+        {
+            this.HuaweiCloudBlobNameCalculator = HuaweiCloudBlobNameCalculator;
```
Comment (rqx110) [inline framework/src/Volo.Abp.BlobStoring.HuaweiCloud/HuaweiCloudBlobProvider.cs]:
```suggestion
            HuaweiCloudBlobNameCalculator = huaweiCloudBlobNameCalculator;
```

File: framework/src/Volo.Abp.BlobStoring.HuaweiCloud/HuaweiCloudBlobProvider.cs [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                {
+                    BucketName = containerName,
+                    Key = blobName,
+                    InputStream = args.BlobStream
+                }
+            );
+
+
+
+
```
Comment (rqx110) [inline framework/src/Volo.Abp.BlobStoring.HuaweiCloud/HuaweiCloudBlobProvider.cs:53]: remove these empty lines, also in other files

File: framework/src/Volo.Abp.BlobStoring.HuaweiCloud/HuaweiCloudBlobNamingNormalizer.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+using System.Text.RegularExpressions;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.Localization;
+
+namespace Volo.Abp.BlobStoring.HuaweiCloud
+{
+    public class HuaweiCloudBlobNamingNormalizer: IBlobNamingNormalizer, ITransientDependency
+    {
+        /// <summary>
+        ///https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html
```
Comment (realLiangshiwei) [inline framework/src/Volo.Abp.BlobStoring.HuaweiCloud/HuaweiCloudBlobNamingNormalizer.cs:14]: https://support.huaweicloud.com/intl/en-us/ugobs-obs/obs_41_0009.html#section2

**Comment by realLiangshiwei** [comment]
Can you add some unit tests?

**Comment by vincywindy** [comment]
> Can you add some unit tests?

I hope so,but Huawei OBS service is not free ~.~!

**Comment by vincywindy** [comment]
> > Can you add some unit tests?
> 
> I hope so,but Huawei OBS service is not free ~.~!

But I use this in my project,it works fine!

**Pull request closed without merging**